Jenkinsfile

```diff
@@ -1,216 +1,64 @@
 #!groovy
 
-@Library('github.com/mozmar/jenkins-pipeline@20170214.1')
+@Library('github.com/mozmar/jenkins-pipeline@20170315.1')
 
-def config
-def utils
-
-stage ('Checkout') {
-    node {
-        checkout scm
-        sh 'git submodule sync'
-        sh 'git submodule update --init --recursive'
-        // clean up
-        sh 'make clean'
-        // load the config
-        config = readYaml file: 'jenkins.yml'
-        // load the utility functions used below
-        utils = load 'jenkins/utils.groovy'
-        // defined in the Library loaded above
-        setGitEnvironmentVariables()
-        setConfigEnvironmentVariables(config)
-        // save the files for later
-        stash name: 'scripts', includes: 'bin/,docker/'
-        stash name: 'tests', includes: 'tests/,requirements/'
-        stash 'workspace'
+def loadBranch(String branch) {
+    // load the utility functions used below
+    utils = load 'jenkins/utils.groovy'
+    is_demo_branch = ( env.BRANCH_NAME ==~ /^demo\/[\w-]+$/ )
```

Comment:
Agreed. The deprecation will be tough, but I don't think it'll be because of the branch names, I think it'll be the app names that will be the issue. Unless we can upgrade Deis soon and go back to the old naming scheme I think a lot of demo URLs that perhaps other people outside our team are watching will cease to be updatable if we merge this. So I think perhaps it should be blocked on that Deis upgrade and I should remove the app renaming and 24 character check. You have thoughts on that?